### PR TITLE
[Merged by Bors] - feat: additional version of FTC-2 for improper integrals

### PR DESCRIPTION
**Pull request by fpvandoorn**
* The extra import doesn't import any additional files transitively.
* From the Sobolev project

Co-authored-by: Heather Macbeth 25316162+hrmacbeth@users.noreply.github.com

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

File: Mathlib/MeasureTheory/Integral/IntegralEqImproper.lean [hunk 1]
```diff
@@ -721,6 +722,32 @@ theorem integrableOn_Ioi_deriv_of_nonneg (hcont : ContinuousOn g (Ici a))
       exact g'pos _ hy.1
 #align measure_theory.integrable_on_Ioi_deriv_of_nonneg MeasureTheory.integrableOn_Ioi_deriv_of_nonneg
 
+/-- **Fundamental theorem of calculus-2**, on semi-infinite intervals `(-∞, a)`.
+When a function has a limit `m` at `-∞`, and its derivative is integrable, then the
+integral of the derivative on `(-∞, a)` is `f a - m`. Version assuming differentiability
+on `(-∞, a)` and continuity on `(-∞, a]`.-/
+theorem integral_Iio_of_hasDerivAt_of_tendsto (hcont : ContinuousOn f (Iic a))
```
Comment (ocfnash) [inline Mathlib/MeasureTheory/Integral/IntegralEqImproper.lean:729]: Can this be weakened to:
```suggestion
theorem integral_Iio_of_hasDerivAt_of_tendsto (hcont : Tendsto f (𝓝[<] a) (𝓝 (f a))
```
(might not be worth the effort)

**Review by ocfnash — APPROVED** [review]
Do you think it's worth adding corresponding results for `Ioi`?

Either way, this LGTM.

bors d+

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**